From 55f8c1a2578a5e4bbd036e9a8c653c4de299940f Mon Sep 17 00:00:00 2001
From: Lothar Rubusch <lothar.rubusch@enclustra.com>
Date: Sun, 27 Nov 2022 20:27:39 +0000
Subject: [PATCH] env: qspi boot - avoid ubifs

Since vivado/petalinux 2022.1 u-boot does not simply source
a bootscript, but prepares a full QSPI boot expecting ubifs
as flash filesystem.

The patch brings reverts the behavior to the legacy one,
source a bootscript.

Note: Probably the better approach would be really to setup
a loadable ubifs filesystem in qspi flash instead of
loading just a ramfs image.

Signed-off-by: Lothar Rubusch <lothar.rubusch@enclustra.com>
---
 include/configs/xilinx_zynqmp.h | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/include/configs/xilinx_zynqmp.h b/include/configs/xilinx_zynqmp.h
index 4bd48c7611..fd8672c442 100644
--- a/include/configs/xilinx_zynqmp.h
+++ b/include/configs/xilinx_zynqmp.h
@@ -133,10 +133,7 @@
 
 #define BOOTENV_DEV_QSPI(devtypeu, devtypel, instance) \
 	"bootcmd_" #devtypel #instance "=sf probe " #instance " 0 0 && " \
-		       "setenv mtdids 'nor0=nor0' && " \
-		       "setenv mtdparts 'mtdparts=nor0:16m(raw),-(boot)' && " \
-		       "mtdparts && " \
-		       "ubi part boot; ubifsmount ubi0:boot; ubifsload $scriptaddr boot.scr; && " \
+		       "sf read ${scriptaddr} ${script_offset_f} ${script_size_f} && " \
 		       "echo QSPI: Trying to boot script at ${scriptaddr} && " \
 		       "source ${scriptaddr}; echo QSPI: SCRIPT FAILED: continuing...;\0"
 
-- 
2.25.1

